report/2.Analysis/ArduinoInEducation.tex

```diff
@@ -0,0 +1,50 @@
+Most of the students expressed that they have no real interest in the syntax. In fact a substantial part indicated that they did not understand what the syntax of a programming language actually means. Below is a short summarization of the most problematic aspects of general syntax and semantic of the arduino language (or any reference language the students might have), in arbitary order.
+
+\textbf{Data Types}\\
+Data types in general, seems to cause alot of problems for the students. Most students can name the different types but far less have the proper knowledge to identify and explain the meaning behind them. The students in most cases do not understand why they have to state the data type explicitly when declaring a variable. When presented with a small example program utilizing type inference on all declarations, the students in most cases, instantly understood how to identify a declaration and how to use the variable properly.
+
+\textbf{Function Call and Definition}\\
+The students concluded that the manner in which C++ and C{\#} makes function/method calls and definitions are somewhat ambigues to them. In relation to syntax they expressed some minor frustrations concerning the overhead nessacery to define simple methods, such as methods that does not return a value and also when making function calls with no arguments. The most problematic aspect of abstraction in general was the matter of scoping. No students expressed any real knowledge of scoping. This is especiallly a problem when it comes to understanding error messages from the compiler. Clearly scoping rules must be kept in mind when developing a new language for beginners.
+
+\textbf{Control Structures}\\
+In general the students understood what the different controlstructures do. But almost non of them could remember the specific syntax and meaning behind them. The selective control structures seems to be easier to understand than the iterative control structures. A substantial amount of the students stated that the control structes in general are difficult to understand and remember, since they do not reflect anything else in the language. At least according to them.
+Clearly the students needs more consistensy in the language. The control structures should in some way mirror the syntax of another major aspect of the languange, to make it more intuitive to use and remember.
+
+\textbf{Operator Precedence and Association}\\
+Both terms were completely forign to the students. Their perception of precedence were mostly based on arithmetic notation. This is sufficient in most cases. The biggest concern was with association. Most of students did not understand that different operators could have different association. which proved to be a major issue when presented with code. In most cases the students understood what part of the expression was calculated first. But choose the wrong anwser based on mostly random factors. Clearly association needs to be as intuitive as possible.
```

Comment:
Change "but choose the wrong anwser" to "but chose the wrong answer" (Both spelling and an o too much)